pkg/resource/function/hooks.go

```diff
@@ -346,3 +353,59 @@ func customPreCompare(
 		}
 	}
 }
+
+// updateFunctionConcurrency calls UpdateFunctionConcurrency to update a specific
+// lambda function reserved concurrent executions.
+func (rm *resourceManager) updateFunctionConcurrency(
+	ctx context.Context,
+	desired *resource,
+) error {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.updateFunctionConcurrency")
+	defer exit(err)
+
+	dspec := desired.ko.Spec
+	input := &svcsdk.PutFunctionConcurrencyInput{
+		FunctionName: aws.String(*dspec.Name),
+	}
+
+	if desired.ko.Spec.ReservedConcurrentExecutions != nil {
+		input.ReservedConcurrentExecutions = aws.Int64(*desired.ko.Spec.ReservedConcurrentExecutions)
+	} else {
+		input.ReservedConcurrentExecutions = aws.Int64(0)
+	}
+
+	_, err = rm.sdkapi.PutFunctionConcurrencyWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "PutFunctionConcurrency", err)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// setResourceAdditionalFields will describe the fields that are not return by
+// GetFunctionConcurrency calls
+func (rm *resourceManager) setResourceAdditionalFields(
+	ctx context.Context,
+	r *resource,
```

Comment:
why do you need both params r & ko ?
anytime you need ko, you can just do "r.ko"

Reply:
Good catch, i initially copied this function for ecr-controller.
Keeping `ko v1alpha1.Function` since the passed `ko` is a deep copy of `r.ko`